WireDomain/Sources/WireDomain/Event Processing/TeamEventProcessor/TeamMemberUpdateEventProcessor.swift

```diff
@@ -32,9 +33,24 @@ protocol TeamMemberUpdateEventProcessorProtocol {
 
 struct TeamMemberUpdateEventProcessor: TeamMemberUpdateEventProcessorProtocol {
 
-    func processEvent(_: TeamMemberUpdateEvent) async throws {
-        // TODO: [WPB-10186]
-        assertionFailure("not implemented yet")
+    enum Error: Swift.Error, Equatable {
+        case failedToFindMember(_ membershipId: UUID)
+    }
+
+    let context: NSManagedObjectContext
```

Comment:
To align with what John did here for the RenameEvent [here](https://github.com/wireapp/wire-ios/commit/f3b06e8f640e3e8eb1e66ad6047e103ede1c7886#diff-a6cf300bceb56a8dcc352452381cf8121e41c3aec7b3687b96df76c5ec742799), we are passing the context directly to the processor however since we have repositories that are responsible of fetching / storing data locally, I wonder if we should be passing repositories instead of context.

Reply:
Abstracting Core Data has its own difficulties, for example the requirement that entities must only be accessed from the context's queue. Since the protocol probably won't allow you to specify this requirement you would need to declare any method async.
That, however, has it's own side effects, for example that you cannot batch several calls inside one `context.perform`.

Generally I think we should fight less against Apple's SDK. I'm not even a great fan of applying MVVM for UIKit, btw.

Reply:
@jullianm wasn't it because the repository was not ready here ?

Reply:
I guess so. I just pushed some changes to use the dedicated `TeamRepository` instead of the context.